src/zeek.bif

```diff
@@ -3399,9 +3399,9 @@ const char* conn_id_string(zeek::Val* c)
 function dump_current_packet%(file_name: string%) : bool
 	%{
 	const Packet* pkt;
+	auto* pkt_src = static_cast<zeek::iosource::PktSrc*>(zeek::run_state::detail::current_iosrc);
```

Comment:
Is this guaranteed to succeed, or could the current IO source be a non-packet source? Do. we need a dynamic_cast to check?

Reply:
The current IO source potentially be a non-packet source, most likely. We previously checked for whether the global was null so it'd be best for that to be dynamic_cast.